GB-Cloud

KuznetsovMK opened the pull request.

Commit:
GB-Cloud-les3 HM

Commit:
gb-cloud#2

vyrexd commented on cloud-storage-oct-2021-client/src/main/java/com/geekbrains/io/App.java:
```diff
@@ -0,0 +1,16 @@
+public class App extends Application {
+    @Override
+    public void start(Stage primaryStage) throws Exception {
+        Parent parent = FXMLLoader.load(getClass().getResource("chat.fxml"));
```
cloud-main.fxml )))

vyrexd commented on cloud-storage-oct-2021-client/src/main/java/com/geekbrains/io/ChatController.java:
```diff
@@ -0,0 +1,183 @@
+
+        else if (msg.split("\n")[0].equals("serverFileInfo")) {
+            input.clear();
+            input.setText(msg.split("\n")[1]);
```
Пайплайн на сериализации, давайте этим пользоваться и передавать данные в свойства класса

vyrexd commented on cloud-storage-oct-2021-server/src/main/java/com/geekbrains/netty/MessageHandler.java:
```diff
@@ -0,0 +1,139 @@
+                localFilePath = Paths.get("localroot").resolve(fileName);
+
+                InputStream is = new FileInputStream(String.valueOf(srvFilePath));
+                OutputStream os = new FileOutputStream(String.valueOf(localFilePath));
```
Не должны быть в посылке пути клиента

Commit:
Добавленный функционал:

1. Download /  upload  / delete
2. Зайти в папку / выйти из папки
3. Переименовать папку / файл
4. Авторизация / Регистрация (sqlite)

Commit:
Обновлено

Добавлено рекурсивное удаление сложенных папок.

Commit:
настроил ломбок

Commit:
update:

add MenuBar -> logout.

Commit:
Merge branch 'gb-cloud-workbranch' into GB-Cloud

Commit:
Update:

1. Добавлен просмотр файлов.
2. Добавлена возможность выбрать локальную папку для переноса файлов на сервер.
3. Добавлена возможность выбрать локальную папку для загрузки файла с сервера.
4. Визуальное изменение клиента.

Commit:
Update:

1. Добавлено открытие файлов через контекстное меню на серверной стороне.

Commit:
Merge branch 'gb-cloud-workbranch' into GB-Cloud

vyrexd commented on cloud-core-oct-2021/src/main/java/com/geekbrains/model/ListMessage.java:
```diff
@@ -0,0 +1,46 @@
+
+        if (dir != null) {
+            setType(Operation.SERVER_LIST_MESSAGE);
+
```
Files.list(p).sorted(Files::isDirectory).map(p -> p.getFileName().toString()).collect(Collectors.toList()).....
